File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        int num = nums[0], cnt = 0, max = 0, ans = 0;
        for(int i : nums){
            if(i == num) cnt++;
            else{
                num = i;
                cnt = 1;
            }
            if(cnt > max){
                ans = i;
                max = cnt;
            }
        }
        return ans;
    }
}

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let idx = 0, cnt = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[idx] == nums[i]) cnt++;
        else cnt--;
        if (cnt == 0) {
            idx = i;
            cnt++;
        }
    }
    return nums[idx];
};